### PR TITLE
[11.0][add][currency_rate_update_vn_VCB]

### PR DESCRIPTION
**Pull request by denislour**

### CODE REVIEW

File: currency_rate_update_VN_VCB/README.rst
Comment (tungocbui) [inline currency_rate_update_VN_VCB/README.rst:47]: You might want to keep only your name in the list for now. Because i don't think other guys contribute in this module

**Comment by tungocbui** [comment]
@denislour tks for your pull request...

File: currency_rate_update_VN_VCB/README.rst
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/README.rst]: I think the usage should explain what is different between the "parent" module from  the currency repository and this module.

**Review by jcdrubay — APPROVED** [review]

File: currency_rate_update_VN_VCB/README.rst
Comment (novawish) [inline currency_rate_update_VN_VCB/README.rst:26]: You just need to add the link to the `currency_rate_update`. No need to repeat the description.

File: currency_rate_update_VN_VCB/__manifest__.py
Comment (novawish) [inline currency_rate_update_VN_VCB/__manifest__.py]: If it's empty, you can remove it

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (novawish) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py]: You can return directly the dictionary of result: 
```
return {'rate_currency': float(
            dom.xpath(xpath_rate_currency, namespaces=ns)[0])}
```

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (novawish) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py]: Extra useless space at the beginning of the docstring

File: currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py
Comment (novawish) [inline currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: No need to search, you just need to call `self.usd.rate_ids.unlink()`

File: currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py
Comment (novawish) [inline currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: Keep only one function `check_currency_rate_update_base_currency` 
```
self.assertEqual(len(self.eur.rate_ids), 0)
self.assertTrue(len(self.usd.rate_ids) > 0)
self.assertTrue(len(self.vnd.rate_ids) > 0)
```
One more function where we change the company's currency: `check_currency_rate_update_new_currency`
```
self.update_service.company_id.currency_id = self.vnd.id
self.assertTrue(len(self.eur.rate_ids) > 0)
self.assertTrue(len(self.usd.rate_ids) > 0)
self.assertEqual(len(self.vnd.rate_ids), 0)
```

**Review by novawish — CHANGES_REQUESTED** [review]

File: currency_rate_update_VN_VCB/README.rst
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/README.rst]: This module does not change usage, but only adds an additional source of exchange rates when configuring in: 
Accounting > Configuration > Multi-currencies > Rate Auto-download

Usage is the same as the module `Currency Rate Update <...>`.

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py]: We should only need the content of the else.
The inverted part is handled by the module currency_rate_inverted and the currency_rate_update (https://github.com/OCA/currency/blob/11.0/currency_rate_update/models/currency_rate_update.py#L177)

File: currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: You need to rename the function to `test_...` instead of `check_...` so the test will be executed.

File: currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: In this function the first step should be to run the service: `self.update_service.refresh_currency()`

File: currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: After updating the company_id.currency_id, you need to run the service.

Comment (jcdrubay) [reply currency_rate_update_VN_VCB/tests/test_currency_rate_update_vn_vcb.py]: `self.update_service.refresh_currency()`

**Review by jcdrubay — CHANGES_REQUESTED** [review]
Still few adjustments required.

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (hieulucky111) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py:63]: Please check again:
`main_curr_data`:  Maybe not available

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (hieulucky111) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py:59]: Maybe, you can add the new information on ReadMe. Or can you add the else statement in here.

**Comment by denislour** [comment]
@novawish @jcdrubay Please check it again :)

**Comment by denislour** [comment]
@jcdrubay , @novawish , @hieulucky111 , Please help me to review it.

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (denislour) [reply currency_rate_update_VN_VCB/services/update_service_VN_VCB.py:59]: @hieulucky111 Actually we don't need `else`statement

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (denislour) [reply currency_rate_update_VN_VCB/services/update_service_VN_VCB.py:63]: @hieulucky111 I checked it many times and we always `main_curr_data` is valid.

File: currency_rate_update_VN_VCB/README.rst
Comment (denislour) [reply currency_rate_update_VN_VCB/README.rst:26]: @novawish I added the new content of `usage` and added the link `currency_rate_update`.

**Comment by denislour** [comment]
@jcdrubay please check it again. I fixed it.

File: currency_rate_update_VN_VCB/__manifest__.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/__manifest__.py]: Should be: 
    "version": "11.0.1.0.0",

https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py

Comment (jcdrubay) [inline currency_rate_update_VN_VCB/__manifest__.py]:
```suggestion
    'version': '11.0.1.0.0',
```

File: currency_rate_update_VN_VCB/__manifest__.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/__manifest__.py]: We should depend on currency_rate_inverted

Comment (jcdrubay) [reply currency_rate_update_VN_VCB/__manifest__.py]: We should force inverted rate for all currencies of the supported_currency_array

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (jcdrubay) [inline currency_rate_update_VN_VCB/services/update_service_VN_VCB.py]: If we use inverted currency, then we don't need anymore to do the division, we will have directly rate=curr_data['rate_currency']

**Review by jcdrubay — CHANGES_REQUESTED** [review]

**Review by jcdrubay — APPROVED** [review]

**Review by hieulucky111 — APPROVED** [review]

File: currency_rate_update_VN_VCB/services/update_service_VN_VCB.py
Comment (denislour) [reply currency_rate_update_VN_VCB/services/update_service_VN_VCB.py]: Done.

File: currency_rate_update_VN_VCB/__manifest__.py
Comment (denislour) [reply currency_rate_update_VN_VCB/__manifest__.py]: I custom a little bit model `currency.rate.update.service` in `new commit` to force inverted rate for all currencies of the supported_currency_array. Please check it.

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [inline currency_rate_update_VN_VCB/models/currency_rate_update.py]: You can have a list of `currencies` inside for loop and then you only have to call `write` once outside of the loop

File: currency_rate_update_VN_VCB/__manifest__.py
Comment (novawish) [inline currency_rate_update_VN_VCB/__manifest__.py]: I think `currency_rate_update` already depends on `base`, so you don't need to depend on it

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [inline currency_rate_update_VN_VCB/models/currency_rate_update.py]: What if the user change the service?

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [inline currency_rate_update_VN_VCB/models/currency_rate_update.py]: `or currency.rate_inverted == True:`

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [inline currency_rate_update_VN_VCB/models/currency_rate_update.py]: `and 'currency_to_update' in vals:`

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [reply currency_rate_update_VN_VCB/models/currency_rate_update.py]: You need to call `fore_inverted_currency` after write

File: currency_rate_update_VN_VCB/models/currency_rate_update.py
Comment (novawish) [inline currency_rate_update_VN_VCB/models/currency_rate_update.py]: `force_inverted_currency`

**Review by novawish — APPROVED** [review]
Looks good to me

**Comment by jcdrubay** [comment]
@OCA/core-maintainers First Pull request to merge in this repository. I am about to click on the Merge button. Do you have any recommendation? Should we Squash commits before merging and have a single commit "[ADD] currency_rate_update_vn_VCB" ?
Thanks in advance for your advice.

**Review by moylop260 — APPROVED** [review]
LGTM :+1:
Just a small comment, the name of python file and folders should be `snake_case` (lowercase and underscore, without uppercases)

But it is not a blocking point.
You can merge if you are agree.

**Pull request merged**
